opencog/atomspace/TypeIndex.cc

```diff
@@ -48,6 +48,31 @@ void TypeIndex::removeAtom(AtomPtr a)
 	remove(t, a->getHandle());
 }
 
+size_t TypeIndex::getNumAtomsOfType(Type type, bool count_subclasses) const
+{
+	iterator it(type, count_subclasses);
+	size_t atom_count = 0;
+	it.s = idx.begin();
+	it.send = idx.end();
+
+	// Loop over all the types looking for type and optional subclasses.
+	Type current_type = 0;
+	while (it.s != it.send)
+	{
+		// If this type is a match...
+		if ((type == current_type) || 
+		    (count_subclasses && (classserver().isA(type, it.type))))
+		{
+			// Add the size of the atom set for this type.
+			atom_count += idx.at(current_type).size();
+		}
+		current_type++;
+		++it.s;
+	}
+
```

Comment:
Yes, that seems like a good idea.  Actually, I think that the classwerver lready has a vector of subclasses in it, and that isA uses that vector. Not sure.  So it would need to be exposed in the API.

Right now, the number of types is small and static.  There were recent discussions on the mailing list (about 2 months ago) which made a rather reasonable argument that, during runtime, there could be many millions of atom types autogenerated.  Basically, every EvaluationLink+PredicateNode would be converted to a new link type, probably inheriting from EvaluationLink.  The classerver would need some work if/when this were to ever happen.